src/lwc/geReviewDonations/geReviewDonations.js

```diff
@@ -13,22 +15,54 @@ export default class geReviewDonations extends NavigationMixin(LightningElement)
     CUSTOM_LABELS = geLabelService.CUSTOM_LABELS;
 
     @api donorId;
-    @api dedicatedListenerEventName = 'geDonationMatchingEvent';
-    @api selectedDonation;
-    @api opportunities;
 
-    @track donationType;
-    @track donor;
+    _donor;
+    _donorType;
+    _donationType;
+    _selectedDonation;
+    _opportunities = [];
+    _hasPreviouslySelectedDonation = false;
+    _dedicatedListenerEventName = 'geDonationMatchingEvent';
+
+    connectedCallback() {
+        registerListener(this._dedicatedListenerEventName, this.handleReceiveEvent, this);
+        registerListener('resetReviewDonationsEvent', this.handleResetReviewDonationsComponent, this);
+    }
+
+    disconnectedCallback() {
+        unregisterListener(this._dedicatedListenerEventName, this.handleReceiveEvent, this);
+        unregisterListener('resetReviewDonationsEvent', this.handleResetReviewDonationsComponent, this);
+    }
 
     @wire(getRecord, { recordId: '$donorId', optionalFields: ['Account.Name', 'Contact.Name'] })
-    wiredGetRecordMethod({ error, data }) {
+    wiredGetRecord({ error, data }) {
+        if (error) return handleError(error);
         if (data) {
-            this.donor = data;
-        } else if (error) {
-            handleError(error);
+            this._donor = data;
+            this._donorType = this._donor.apiName;
         }
     }
 
+    @wire(getOpenDonations, { donorId: '$donorId', donorType: '$_donorType' })
+    wiredGetOpenDonations({ error, data }) {
+        if (error) return handleError(error);
+        if (data) return this.opportunities = JSON.parse(data);
+    }
+
+    set opportunities(value) {
+        if (!value) return this._opportunities = [];
```

Comment:
I don't think a setter should return anything.  Can this._opportunities be defaulted to [] to avoid this line?

Reply:
Updated!